controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,873 @@
+	}
+	if config.Spec.TenantID == "" {
+		return fmt.Errorf(cannotBeNilError, "tenantId", config.ClusterName)
+	}
+	if config.Spec.AzureCredentialSecret == "" {
+		return fmt.Errorf(cannotBeNilError, "azureCredentialSecret", config.ClusterName)
+	}
+	clientId, clientSecret, err := getSecrets(h.secretsCache, config.Spec.AzureCredentialSecret)
+	if err != nil {
+		return fmt.Errorf("could not get secrets with error: %v", err)
+	}
+	if clientId == "" || clientSecret == "" {
+		return fmt.Errorf(cannotBeNilError, "clientId and clientSecret", config.ClusterName)
+	}
+	if config.Spec.Imported {
+		return nil
+	}
+	if config.Spec.BaseURL == "" {
+		return fmt.Errorf(cannotBeNilError, "baseUrl", config.ClusterName)
+	}
+	if config.Spec.AuthBaseURL == "" {
+		return fmt.Errorf(cannotBeNilError, "authBaseUrl", config.ClusterName)
+	}
+	if config.Spec.KubernetesVersion == "" {
+		return fmt.Errorf(cannotBeNilError, "kubernetesVersion", config.ClusterName)
+	}
+
+	azureAuthorizer, err := newClientAuthorizer(h.secretsCache, config.Spec)
+	if err != nil {
+		return err
+	}
+	serviceClient, err := newServiceClient(azureAuthorizer, config.Spec)
+	if err != nil {
+		return err
+	}
+	listOperators, err := serviceClient.ListOrchestrators(ctx, config.Spec.ResourceLocation, "managedClusters")
+	if err != nil {
+		return err
+	}
+	objOrchestrators := *listOperators.Orchestrators
+	validKubernetesVersion := false
+	for _, obj := range objOrchestrators {
+		if config.Spec.KubernetesVersion == to.String(obj.OrchestratorVersion) {
+			validKubernetesVersion = true
+			break
+		}
+	}
+	if !validKubernetesVersion {
+		return fmt.Errorf("field kubernetesVersion is invalid for non-import [%s] cluster config", config.ClusterName)
+	}
+	if hasAgentPoolProfile(config.Spec) {
+		for _, np := range config.Spec.NodePools {
+			if np.Name == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Name", config.ClusterName)
+			}
+			if np.Count == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Count", config.ClusterName)
+			}
+			if np.MaxPods == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.MaxPods", config.ClusterName)
+			}
+			if np.VMSize == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.VMSize", config.ClusterName)
+			}
+			if np.OsDiskSizeGB == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OsDiskSizeGB", config.ClusterName)
+			}
+			if np.OsDiskType == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OSDiskType", config.ClusterName)
+			}
+			if np.Mode == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Mode", config.ClusterName)
+			}
+			if np.OsType == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OsType", config.ClusterName)
+			}
+		}
+	}
+	if (config.Spec.PublicAccess && config.Spec.PrivateAccess) ||
+		(!config.Spec.PublicAccess && !config.Spec.PrivateAccess) {
+		return fmt.Errorf("only one field PublicAccess or PrivateAccess have to be set for [%s] cluster config", config.ClusterName)
```

Comment:
They both should have a value, but only one can be true. Although this is in the case of EKS. I don't actually see the PublicAccess and privateAccess fields used anywhere. Are these needed for aks? Could someone configure something similar just by setting `AuthorizedIPRanges`?

Reply:
These may be left over from EKS? I am not sure these are required for AKS. For AKS, by default, the public endpoint for the APIserver is on unless you set up a private cluster in which case the public access is turned off. The `AuthorizedIPRanges` is a separate setting for restricted IP.

I don't see any other code setting related to private clusters; maybe we haven't implemented it yet. 

<img width="327" alt="Screen Shot 2021-03-03 at 4 46 02 PM" src="https://user-images.githubusercontent.com/6218159/109893125-edd6e700-7c3f-11eb-8bec-3df9ab0a8654.png">
access.

Reply:
It is used now, sorry I have forgotten to add `enablePrivateCluster` option